core/system/sys_stat.go

```diff
@@ -48,15 +56,49 @@ func retrieveAndUpdateSystemStat() {
 		logger.Warnf("Failed to retrieve current system load: %+v", err)
 	}
 	if len(cpuStats) > 0 {
-		// TODO: calculate the real CPU usage
-		// cpuStat := cpuStats[0]
-		// currentCpuUsage.Store(cpuStat.User)
+		curCpuStat := cpuStats[0]
+		recordCpuUsage(prevCpuStat, curCpuStat)
+		// Cache the latest CPU stat info.
+		prevCpuStat = curCpuStat
 	}
 	if loadStat != nil {
 		currentLoad.Store(loadStat.Load1)
 	}
 }
 
+func recordCpuUsage(prev, curCpuStat cpu.TimesStat) {
```

Comment:
In order to avoid struct copy, it might be better to use Pointer instead of struct here?

Reply:
Good idea. I've updated the code and added a `stat.system.collectIntervalMs` conf item. Please check.